babel_monitor/src/lib.rs

```diff
@@ -223,6 +227,38 @@ impl<T: Read + Write> Babel<T> {
         }
         Ok(vector)
     }
+
+    /// In this function we take a route snapshot then loop over the routes list twice
+    /// to find the neighbor local address and then the route to the destination
+    /// via that neighbor. This could be dramatically more efficient if we had the neighbors
+    /// local ip lying around somewhere.
+    pub fn get_route_via_neigh(
+        &mut self,
+        neigh_mesh_ip: IpAddr,
+        dest_mesh_ip: IpAddr,
+    ) -> Result<Route, Error> {
+        let routes = self.parse_routes()?;
+        // First find the neighbors route to itself to get the local address
+        for neigh_route in routes.iter() {
+            // This will fail on v4 babel routes etc
+            if let IpNetwork::V6(ref ip) = neigh_route.prefix {
+                if ip.get_network_address() == neigh_mesh_ip {
+                    let neigh_local_ip = neigh_route.neigh_ip;
+                    // Now we take the neigh_local_ip and search for a route via that
+                    for route in routes.iter() {
```

Comment:
IMO I would break this into 2 sequential loops instead of a nested one (unless there is a particular reason for it)

Reply:
mostly me not wanting to make a placeholder neighbour object to deal with the scoping issue, it was more annoying that it should have been.

Reply:
Ok fair enough